OneCommunityGlobal/HighestGoodNetworkApp
# Jiayu --> Shuhua - All Inventory Types page functionalities

# Description

Implement the functionalities of the All Inventory Types page, which the UI was implemented in this [PR](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1868)

## Related PRS (if any):

This frontend PR is related to the [#765](https://github.com/OneCommunityGlobal/HGNRest/pull/765) backend PR.

## Main changes explained:

- added actions and reduces for updating/deleting inventory types, along with corresponding toast notifications
- added actions and reduces for adding/deleting units of measurement, along with corresponding toast notifications
- added click event handlers for all **Add**, **Edit**, **Delete** buttons

## How to test:

1. check into current branch
2. do `npm run start:local` to run this PR locally
3. Clear site data/cache or use Incognito mode
4. log in as any user, direct to `http://localhost:3000/bmdashboard`, log in again, and navigate to `http://localhost:3000/bmdashboard/inventorytypes`
5. (optional) add new inventory types for testing, check out the note below for adding multiple types at once
6. Verify editing and deleting inventory types
    1. choose any category, then a type
    2. click “**Edit**”, update either name or description, confirm changes and toasts
    3. click “**Delete**”, confirm changes and toasts
7. Verify adding and removing inventory units
    1. click “**Unit of Measurement**”, then “**Add**” to create a new unit, or “**Delete**” to remove it 
    2. confirm changes and toasts
8. Other buttons
    1. "**Add**" buttons under categories open new pages, but right now, only “Add Material” page and “Add Equipment” page are fully functional, the “Add Tool” page is implemented but not connected with the backend
    2. "**Back to previous list page**" does nothing at this point since the target page is not implemented yet

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/83729360/049c15ab-21ed-44be-b330-50874ab7d033

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/83729360/506709d0-ae7d-4d87-b3c2-903e30cbb0cf




## Note:

1. You can use MongoDB Compass to quickly create a new type for each category. Here’s the [JSON](https://drive.google.com/file/d/1Bw4eTm6-1VodeuHHxYUV-RPnVPTwAUAx/view) in case you want to try it out. 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/83729360/fbced6e1-498d-4872-8fee-0289b72a4bfa


2. These potential errors are unrelated
<img width="1310" alt="potential errors" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/83729360/1fcfd9d3-3139-4a64-b04f-83e83453b0fd">
